### PR TITLE
test prtest image update

### PR DESCRIPTION
**Pull request by zhulinJulia24**
1. update image for pr test
2. add vl pipeline testcases
3. restful server test only run memory_testcase to make pr more stable

### CODE REVIEW

File: .github/workflows/pr_ete_test.yml [hunk 1]
```diff
@@ -43,36 +43,40 @@ jobs:
     steps:
       - name: Setup systems
         run: |
-          rm /etc/apt/sources.list.d/cuda*.list
-          apt-get update && apt-get install -y --no-install-recommends rapidjson-dev \
-              libgoogle-glog-dev libgl1 openjdk-8-jre-headless
-          rm -rf /var/lib/apt/lists/*
+          rm /etc/apt/sources.list.d/cuda*.list && apt-get update -y && apt-get install -y software-properties-common wget vim &&\
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:46]: this is hard to view, could we separate this long commands into multiple lines?

File: .github/workflows/pr_ete_test.yml [hunk 2]
```diff
@@ -43,36 +43,40 @@ jobs:
...
+          python3 -m pip install /root/packages/flash_attn-2.6.3+cu123torch2.3cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
       - name: Build lmdeploy
         run: |
           python3 -m pip install cmake
           python3 -m pip install -r requirements/build.txt
+          wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.5.tar.gz
+          tar xf openmpi-4.1.5.tar.gz && cd openmpi-4.1.5 && ./configure --prefix=/usr/local/openmpi
+          make -j$(nproc) && make install && cd .. && rm -rf openmpi-4.1.5*
+          export PATH=$PATH:/usr/local/openmpi/bin
+          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/openmpi/lib
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:67]: make it changed for the following steps
```
echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH"  >> $GITHUB_ENV
```

File: .github/workflows/pr_ete_test.yml [hunk 3]
```diff
@@ -43,36 +43,40 @@ jobs:
...
           mkdir build
           cd build
-          cp -r /nvme/qa_test_models/offline_pkg/_deps .
           cmake .. \
               -DCMAKE_BUILD_TYPE=RelWithDebInfo \
               -DCMAKE_EXPORT_COMPILE_COMMANDS=1 \
               -DCMAKE_INSTALL_PREFIX=/opt/tritonserver \
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:73]: no triton server anymore, we should rename the installed directory.

File: .github/workflows/pr_ete_test.yml [hunk 4]
```diff
@@ -43,36 +43,40 @@ jobs:
...
               -DBUILD_MULTI_GPU=ON \
+              -DBUILD_CUTLASS_MOE=OFF \
+              -DBUILD_CUTLASS_MIXED_GEMM=OFF \
               -DCMAKE_CUDA_FLAGS="-lineinfo" \
-              -DUSE_NVTX=ON \
-              -DSM=80 \
-              -DCMAKE_CUDA_ARCHITECTURES=80 \
-              -DBUILD_TEST=OFF
+              -DUSE_NVTX=ON &&\
           make -j$(nproc) && make install
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:80]: should inclulde lib of lmdeploy
```
export LD_LIBRARY_PATH=/opt/lmdeploy/install/lib:$LD_LIBRARY_PATH
echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH"  >> $GITHUB_ENV
```

File: .github/workflows/pr_ete_test.yml (hunk 2 above)
Comment (zhulinJulia24) [reply .github/workflows/pr_ete_test.yml:67]: > make it changed for the following steps
> 
> ```
> echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH"  >> $GITHUB_ENV
> ```

It seems only used in this step.

File: .github/workflows/pr_ete_test.yml (hunk 1 above)
Comment (zhulinJulia24) [reply .github/workflows/pr_ete_test.yml:48]: make it same as docker file

File: autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py [hunk 5]
```diff
@@ -32,3 +32,20 @@ def test_pipeline_chat_tp2(config, model, worker_id):
...
+
+
+@pytest.mark.pipeline_chat
+@pytest.mark.gpu_num_1
+@pytest.mark.pr_test
+@pytest.mark.parametrize('model', [
+    'liuhaotian/llava-v1.6-vicuna-7b', 'OpenGVLab/InternVL2-4B',
+    'OpenGVLab/InternVL2-8B', 'internlm/internlm-xcomposer2d5-7b'
+])
+def test_pipeline_pr_test(config, model, worker_id):
```
Comment (RunningLeon) [inline autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py:44]: this case is not triggered. something wrong?
```
autotest/tools/quantization/test_quantization_w4a16.py .                 [  6%]
autotest/tools/convert/test_convert.py ..                                [ [20](https://github.com/InternLM/lmdeploy/actions/runs/10244807978/job/28338525428?pr=2192#step:9:21)%]
autotest/tools/pipeline/test_pipeline_chat_pytorch.py .                  [ 26%]
autotest/tools/pipeline/test_pipeline_chat_turbomind.py ..               [ 40%]
autotest/tools/restful/test_restful_chat_hf_turbomind.py ..              [ 53%]
autotest/tools/restful/test_restful_chat_workspace.py ..                 [ 66%]
autotest/tools/chat/test_command_chat_hf_pytorch.py .                    [ 73%]
autotest/tools/chat/test_command_chat_hf_turbomind.py ..                 [ 86%]
autotest/tools/chat/test_command_chat_workspace.py ..                    [100%]
```

Comment (zhulinJulia24) [reply autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py:44]: > this case is not triggered. something wrong?
> 
> ```
> autotest/tools/quantization/test_quantization_w4a16.py .                 [  6%]
> autotest/tools/convert/test_convert.py ..                                [ [20](https://github.com/InternLM/lmdeploy/actions/runs/10244807978/job/28338525428?pr=2192#step:9:21)%]
> autotest/tools/pipeline/test_pipeline_chat_pytorch.py .                  [ 26%]
> autotest/tools/pipeline/test_pipeline_chat_turbomind.py ..               [ 40%]
> autotest/tools/restful/test_restful_chat_hf_turbomind.py ..              [ 53%]
> autotest/tools/restful/test_restful_chat_workspace.py ..                 [ 66%]
> autotest/tools/chat/test_command_chat_hf_pytorch.py .                    [ 73%]
> autotest/tools/chat/test_command_chat_hf_turbomind.py ..                 [ 86%]
> autotest/tools/chat/test_command_chat_workspace.py ..                    [100%]
> ```

image
here. vl testcase use 1 GPU, In order to maximize the use of GPU, the use case is executed in two steps.

File: .github/workflows/pr_ete_test.yml (hunk 3 above)
Comment (RunningLeon) [reply .github/workflows/pr_ete_test.yml:75]: the lib of turbomind is not in `LD_LIBRARY_PATH`

File: autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py (hunk 5 above)
Comment (RunningLeon) [reply autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py:44]: why is it not shown as the first one?

Comment (zhulinJulia24) [reply autotest/tools/pipeline/test_pipeline_chat_turbomind_vl.py:44]: > why is it not shown as the first one?

-n 2 which means run in parallel is used. By using -n test process show like this.

File: .github/workflows/pr_ete_test.yml [hunk 6]
```diff
@@ -33,58 +34,53 @@ jobs:
...
+        - /mnt/187:/mnt/187
         - /usr/share/zoneinfo/Asia/Shanghai:/etc/localtime:ro
     steps:
       - name: Setup systems
         run: |
-          rm /etc/apt/sources.list.d/cuda*.list
-          apt-get update && apt-get install -y --no-install-recommends rapidjson-dev \
-              libgoogle-glog-dev libgl1 openjdk-8-jre-headless
-          rm -rf /var/lib/apt/lists/*
+          rm /etc/apt/sources.list.d/cuda*.list && apt-get update -y && apt-get install -y software-properties-common wget vim &&\
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:48]: ninja is not installed

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**